homeassistant/components/reolink/host.py

```diff
@@ -359,6 +365,34 @@ def unregister_webhook(self):
         webhook.async_unregister(self._hass, self.webhook_id)
         self.webhook_id = None
 
+    async def _poll_all_motion(self, *_) -> None:
+        """Poll motion and AI states until the first ONVIF push is received."""
+        if self._webhook_reachable.is_set():
+            # ONVIF push is working, stop polling
+            self._cancell_poll = None
+            return
+
+        try:
+            await self._api.get_motion_state_all_ch()
```

Comment:
I think it would be better to do ONVIF PullPoints instead since they are effectively push via a long poll as the POST resolves as soon as there is a new data and you won't poll too often unless events are actually happening

Reply:
Implementing the ONVIF PullPoint will require some significant changes/additions to the upstream reolink-aio library.

@bdraco is it maybe possible to merge this PR as is with the HTTP polling while I start working on inplementing ONVIF PullPoint long polling in a follow up PR?

Reply:
I think it would be better to wait to have a proper pull point implementation as the trade off with this solution as implemented is that it will frequent polling and the lowest we usually allow is 5s intervals.

Reply:
You can likely lift a huge chunk of the code from 
https://github.com/hunterjm/python-onvif-zeep-async/blob/a153df1dfc3a5877d1ad71ad651992287a1a55be/onvif/managers.py#L273
https://github.com/home-assistant/core/blob/3a846a82ac5a5e4dba49b9378352716348ee3c07/homeassistant/components/onvif/event.py#L225

Reply:
@bdraco I have finally finished implementing ONVIF long polling for the Reolink integration as you recommended.
Thank you very much for pointing me in the right direction.
The PR for it is here: https://github.com/home-assistant/core/pull/94770

I have tested it extensively and on my cams it works very nicely.
Smooth transitions between ws base notification, long polling and fast polling.
Recovers itself after network outage etc.